### ISSUE TITLE
Recovery Code Usage

### ISSUE THREAD

**Issue by barnett** [Houdini/two_factor_authentication#98]
Was unable to find recovery code functionality (like [Github](https://github.com/settings/auth/recovery-codes)). Was seeing if that would fit under the scope of the gem and be ok to add?

If so I'm able to work on/propose an implementation.

[Github Wiki Page on Recovery Codes](https://help.github.com/articles/downloading-your-two-factor-authentication-recovery-codes/)

**Comment by rmm5t** [comment]
Recovery codes are effectively just a 3rd level of password protection (a traditional password is the 1st, and the OTP secret is the 2nd). Personally, I think recover codes are only good for services that have a crazy amount of technically savvy users (e.g. GitHub) to help prevent an overload of customer support requests. 

Most users don't have the right tools in place to securely store recovery codes for proper use possibly years later.  e.g. My parents would have a hard time knowing how to keep a reliable and secure backup of recovery codes if they started to utilize MFA across their services. 

I think there's a bit of a trend for some MFA implementations to just skip them altogether, and allow for a more traditional MFA reset while verifying data via a customer support request. 

In summary, a recovery code is just another randomly generated password, and if you want to support them, it might be best to do that out of band from a MFA gem like this.

**Comment by Houdini** [comment]
Agree with @rmm5t 
@barnett anyway it's possible to create your own recovery code functionality right inside
```ruby
def need_two_factor_authentication?(request)
end
```
